### PR TITLE
created category pages and filter/sort products

### PR DESCRIPTION
**Pull request by Ksenaaa**

### CODE REVIEW

File: src/components/CategoryPage/CategoryPage.tsx [hunk 1]
```diff
@@ -11,15 +30,164 @@ type Props = {
...
+        }
+      },
+      undefined,
+      { shallow: true }
+    )
+    // eslint-disable-next-line
+  }, [page, filter, sort])
+
+  const categoryId = menuCategories.find(categoryElement => categoryElement.path === pageCategory)?.id as string
+  const categoryName = menuCategories.find(categoryElement => categoryElement.path === pageCategory)?.name || ''
```
Comment (vell28) [inline src/components/CategoryPage/CategoryPage.tsx]:
```
const category = menuCategories.find(categoryElement => categoryElement.path === pageCategory)

const categoryId  = category?.id
const categoryName  = category?.name 
```

File: src/components/CategoryPage/CategoryPage.tsx [hunk 2]
```diff
@@ -11,15 +30,164 @@ type Props = {
...
-  const categoryName = menuCategories.find(categoryData => categoryData.path === category)?.name || ''
+  const handleRemoveFilter = useCallback(() => {
+    setFilter(null)
+    setPage(firstNumberPage)
+    setProducts([])
+  }, [])
+
+  const handleRemoveElementFilter = useCallback((element: string) => {
+    setFilter(preState => {
+      const removeElement = Object.fromEntries(Object.entries(preState as IFilterApply)
```
Comment (vell28) [inline src/components/CategoryPage/CategoryPage.tsx]: `Object.key || Object.value` ?

File: src/components/CategoryPage/CategoryPage.tsx [hunk 3]
```diff
@@ -11,15 +30,164 @@ type Props = {
...
   return (
     <div className={styles.wrapper}>
+      {isLoading && <Loader/>}
+
       <div className={styles.title}>
         {t(categoryName)}
       </div>
+
+      <div className={styles.filter}>
+        {isSuccess && data?.pageData.length !== 0 &&
```
Comment (vell28) [inline src/components/CategoryPage/CategoryPage.tsx]: move `data?.pageData.length` to variable

File: src/components/CategoryPage/CategoryPage.tsx [hunk 4]
```diff
@@ -11,15 +30,164 @@ type Props = {
...
+                    handleFilterSortChange={handleFilterSortChange}
+                  />
+                </FilterModal>
+              </div>
+              <Button
+                name="Remove filter"
+                onClick={handleRemoveFilter}
+                isDisabled={!filter}
+              />
+              {filter && Object.entries(filter).map(element =>
```
Comment (vell28) [inline src/components/CategoryPage/CategoryPage.tsx]: as i see you use only key. Why `Object.entries` ?

File: src/components/CategoryPage/FilterElementCheckbox/FilterElementCheckbox.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+type Props = {
+  element: IFilterShowCheckbox,
+  selectedFilter: IFilterApply | null,
+  onCheck: (title: string, value: string) => void
+}
+
+export const FilterElementCheckbox: FC<Props> = ({ element, onCheck, selectedFilter }) => {
+  const handleCheck = (e: MouseEvent<HTMLInputElement>) => {
+    const input = e.target as HTMLInputElement
```
Comment (vell28) [inline src/components/CategoryPage/FilterElementCheckbox/FilterElementCheckbox.tsx:17]: do we really need `as HTMLInputElement`?

File: src/components/CategoryPage/FilterElementCheckbox/FilterElementCheckbox.tsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+    input.value && onCheck(element.title, input.value)
+  }
+
+  let selectedValue = [] as string[]
```
Comment (vell28) [inline src/components/CategoryPage/FilterElementCheckbox/FilterElementCheckbox.tsx]: use useMome here, with condition below

File: src/components/CategoryPage/FilterElementChecked/FilterElementChecked.tsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import React, { FC } from 'react'
+
+import Image from 'next/image'
+
+import removeElement from 'img/svg/close-icon.svg'
+
+import styles from './FilterElementChecked.module.scss'
+
+type Props = {
+  elementChecked: string,
```
Comment (vell28) [inline src/components/CategoryPage/FilterElementChecked/FilterElementChecked.tsx]: checkedElement (Name, id  or what?)

File: src/components/CategoryPage/FilterElementRange/FilterElementRange.tsx [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+type Props = {
+  value: IFilterApply | null,
+  element: IFilterShowRange,
+  onChange: (title: string, value: IRangeFilter) => void
+}
+
+export const FilterElementRange: FC<Props> = ({ value, element, onChange }) => {
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const input = e.target as HTMLInputElement
```
Comment (vell28) [inline src/components/CategoryPage/FilterElementRange/FilterElementRange.tsx:17]: if you need use `as HTMLInputElement` you applied wrong type for `event`

File: src/components/ProductInfo/ProductsByElement/ProductsByElement.tsx [hunk 9]
```diff
@@ -1,26 +1,29 @@
...
 
-export const ProductsByCategory: FC<Props> = ({ category }) => {
-  const { data, isSuccess } = useGetProductsByCategoryQuery({
-    key: category,
+export const ProductsByElement: FC<Props> = ({ element, elementName, titleName }) => {
+  const { data } = useGetProductsByElementQuery({
+    element,
+    elementName,
     page: 1,
     limit: 6
```
Comment (vell28) [inline src/components/ProductInfo/ProductsByElement/ProductsByElement.tsx]: limit = 6?

File: src/utils/helpers/addFilterElement.ts [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    return { ...removeElement }
+  }
+
+  return { ...prevState, [title]: [...preTitle, value] }
+}
+
+export const addFilterRange = (prevState: IFilterApply | null, title: string, value: IRangeFilter) => {
+  if (!prevState) return { [title]: value }
+
+  const removeElement = Object.fromEntries(Object.entries(prevState).filter(element => element[0] !== title))
```
Comment (vell28) [inline src/utils/helpers/addFilterElement.ts]: I'm shure you don't need to use `Object.entries` inside `Object.fromEntries`. I believe exists better way

File: src/utils/helpers/parseParamQuery.ts [hunk 11]
```diff
@@ -0,0 +1,34 @@
+import { IFilterApply, IRangeFilter } from 'interface/filterInterface'
+import { ISortApply } from 'interface/sortInterface'
+
+export const parseParamSortQuery = (param: string | string[] | undefined) => {
```
Comment (vell28) [inline src/utils/helpers/parseParamQuery.ts]: param?: string | string[]

File: src/utils/helpers/parseParamQuery.ts [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+  (param as string).split(';').forEach((element: string) => {
+    const newElement = element.split('=')
+
+    parseParam = { ...parseParam, [newElement[0]]: newElement[1] }
+  })
+
+  return parseParam
+}
+
+export const parseParamFilterQuery = (param: string | string[] | undefined) => {
```
Comment (vell28) [inline src/utils/helpers/parseParamQuery.ts]: param?: string | string[]

File: src/components/CategoryPage/CategoryPage.tsx (hunk 1 above)
Comment (Ksenaaa) [reply src/components/CategoryPage/CategoryPage.tsx]: fixed

File: src/components/CategoryPage/CategoryPage.tsx (hunk 3 above)
Comment (Ksenaaa) [reply src/components/CategoryPage/CategoryPage.tsx]: fixed

File: src/components/CategoryPage/CategoryPage.tsx (hunk 4 above)
Comment (Ksenaaa) [reply src/components/CategoryPage/CategoryPage.tsx]: fixed

File: src/components/CategoryPage/FilterElementChecked/FilterElementChecked.tsx (hunk 7 above)
Comment (Ksenaaa) [reply src/components/CategoryPage/FilterElementChecked/FilterElementChecked.tsx]: fixed

File: src/components/CategoryPage/CategoryPage.tsx (hunk 2 above)
Comment (Ksenaaa) [reply src/components/CategoryPage/CategoryPage.tsx]: fixed

File: src/utils/helpers/parseParamQuery.ts (hunk 11 above)
Comment (Ksenaaa) [reply src/utils/helpers/parseParamQuery.ts]: fixed

File: src/utils/helpers/parseParamQuery.ts (hunk 12 above)
Comment (Ksenaaa) [reply src/utils/helpers/parseParamQuery.ts]: fixed

File: src/components/ProductInfo/ProductsByElement/ProductsByElement.tsx (hunk 9 above)
Comment (Ksenaaa) [reply src/components/ProductInfo/ProductsByElement/ProductsByElement.tsx]: fixed

File: src/components/CategoryPage/FilterElementCheckbox/FilterElementCheckbox.tsx (hunk 5 above)
Comment (Ksenaaa) [reply src/components/CategoryPage/FilterElementCheckbox/FilterElementCheckbox.tsx:17]: discussed

File: src/components/CategoryPage/FilterElementRange/FilterElementRange.tsx (hunk 8 above)
Comment (Ksenaaa) [reply src/components/CategoryPage/FilterElementRange/FilterElementRange.tsx:17]: discussed

File: src/utils/helpers/addFilterElement.ts (hunk 10 above)
Comment (Ksenaaa) [reply src/utils/helpers/addFilterElement.ts]: fixed

File: src/components/CategoryPage/FilterElementCheckbox/FilterElementCheckbox.tsx (hunk 6 above)
Comment (Ksenaaa) [reply src/components/CategoryPage/FilterElementCheckbox/FilterElementCheckbox.tsx]: changed the logic

**Review by vell28 — APPROVED** [review]

**Pull request merged**